### PR TITLE
fix temporal type default value bug mysql

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://github.com/airbytehq/airbyte/issues/15840
The bug is happening for datetime, time and date data types with default values. While building the internal schema of the columns, debezium is failing with the error.

### CODE REVIEW

File: airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java [hunk 1]
```diff
@@ -50,13 +60,33 @@ private void registerDate(final RelationalColumn field, final ConverterRegistrat
...
+            return DateTimeConverter.convertToDate(LocalDate.ofEpochDay((Integer) x));
+          }
+          return DateTimeConverter.convertToDate(x);
+        case "TIME":
+          if (x instanceof Long) {
+            long l = Math.multiplyExact((Long) x, TimeUnit.MICROSECONDS.toNanos(1));
+            return DateTimeConverter.convertToTime(LocalTime.ofNanoOfDay(l));
+          }
+          return DateTimeConverter.convertToTime(x);
+        case "TIMESTAMP":
```
Comment (edgao) [inline airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java:85]: just curious: did you test with different precisions? It seems kind of weird that debezium would have special handling for datetime but not timestamp (and that `timestamp` isn't actually in their table >.> )

File: airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java [hunk 2]
```diff
@@ -50,13 +60,33 @@ private void registerDate(final RelationalColumn field, final ConverterRegistrat
...
-      return switch (fieldType.toUpperCase(Locale.ROOT)) {
-        case "DATETIME" -> DateTimeConverter.convertToTimestamp(x);
-        case "DATE" -> DateTimeConverter.convertToDate(x);
-        case "TIME" -> DateTimeConverter.convertToTime(x);
-        case "TIMESTAMP" -> DateTimeConverter.convertToTimestampWithTimezone(x);
-        default -> throw new IllegalArgumentException("Unknown field type  " + fieldType.toUpperCase(Locale.ROOT));
-      };
+      switch (fieldType.toUpperCase(Locale.ROOT)) {
+        case "DATETIME":
+          if (x instanceof Long) {
```
Comment (edgao) [inline airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java:65]: nitpick: `x instanceof Long l`

**Review by edgao — APPROVED** [review]

**Review by grishick — APPROVED** [review]

File: airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java (hunk 1 above)
Comment (subodh1810) [reply airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java:85]: Yup. They have listed it in the [docs](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-temporal-types) as well

> Excluding the TIMESTAMP data type, MySQL temporal types depend on the value of the time.precision.mode connector configuration property.

**Comment by danieldiamond** [comment]
this worked great, thanks so much for the fast turnaround @subodh1810 !

**Pull request merged**